src/tools/miri/src/shims/unix/linux/sync.rs

```diff
@@ -213,7 +233,7 @@ pub fn futex<'tcx>(
                     break;
                 }
             }
-            this.write_scalar(Scalar::from_target_isize(n, this), dest)?;
+            this.write_scalar(Scalar::from_target_isize(if op != futex_unlock_pi { n } else { 0 }, this), dest)?;
```

Comment:
Does futex_unlock_pi not return how many threads were woken up?

Reply:
> FUTEX_UNLOCK_PI
>   Returns 0 if the futex was successfully unlocked.